codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Connection.java

```diff
@@ -83,9 +85,14 @@
         void onWeightChanged(Http2Stream stream, short oldWeight);
 
         /**
-         * Called when a GO_AWAY frame has either been sent or received for the connection.
+         * Called when a GO_AWAY frame was sent for the connection.
+         */
+        void goAwaySent(long errorCode, ByteBuf debugData);
+
+        /**
+         * Called when a GO_AWAY frame was received for the connection.
          */
-        void goingAway();
+        void goAwayReceived(long errorCode, ByteBuf debugData);
```

Comment:
Agreed, this is a bit hokey :).  The problem is that FrameListener is only used for decoding, so there is no unified way of handling both inbound and outbound GOAWAY. 

Without a common interface for this, the application would have to decorate the FrameWriter in order to intercept the GOAWAY. This isn't horrible I suppose, but if we're going to go with this approach then I think we should get rid of any GOAWAY events in Connection.Listener altogether.  WDYT?